test/e2e/authorization.go

```diff
@@ -711,19 +752,98 @@ var _ = g.Describe("Authorization via admission-controller [RBAC] [Zalando]", fu
+				var err error
+
+				client, err = getCollaboratorClient(eksCluster, awsAccountID)
+				framework.ExpectNoError(err)
+			})
+
+			g.It("should allow write access for non-system resources", func() {
+				err := client.RbacV1().ClusterRoles().Delete(context.Background(), nonSystemResource.Name, metav1.DeleteOptions{DryRun: []string{"All"}})
+				framework.ExpectNoError(err, "failed to delete cluster role: %s", nonSystemResource.Name)
+			})
+
+			g.It("should allow write access for collaborator resources", func() {
+				err := client.RbacV1().ClusterRoles().Delete(context.Background(), collaboratorResource.Name, metav1.DeleteOptions{DryRun: []string{"All"}})
+				framework.ExpectNoError(err, "failed to delete cluster role: %s", collaboratorResource.Name)
+			})
+
+			g.It("should deny write access for system resources", func() {
+				err := client.RbacV1().ClusterRoles().Delete(context.Background(), systemResource.Name, metav1.DeleteOptions{DryRun: []string{"All"}})
+				gomega.Expect(err).To(gomega.MatchError(gomega.ContainSubstring("write operations are forbidden")))
+			})
+
+			// test specific namespaces
+			g.It("should allow deletion of non-system namespace", func() {
+				err := client.CoreV1().Namespaces().Delete(context.Background(), nonSystemResource.Namespace, metav1.DeleteOptions{DryRun: []string{"All"}})
+				framework.ExpectNoError(err, "failed to delete namespace: %s", nonSystemResource.Name)
+			})
+
+			g.It("should deny deletion of visibility namespace", func() {
+				err := client.CoreV1().Namespaces().Delete(context.Background(), collaboratorResource.Namespace, metav1.DeleteOptions{DryRun: []string{"All"}})
+				gomega.Expect(err).To(gomega.MatchError(gomega.ContainSubstring("write operations are forbidden")))
+			})
+
+			g.It("should deny deletion of kube-system namespace", func() {
+				err := client.CoreV1().Namespaces().Delete(context.Background(), systemResource.Namespace, metav1.DeleteOptions{DryRun: []string{"All"}})
+				gomega.Expect(err).To(gomega.MatchError(gomega.ContainSubstring("this namespace may not be deleted")))
+			})
+		})
+
+		g.Context("as engineer user", func() {
+			var client *kubernetes.Clientset
+
+			g.BeforeEach(func() {
+				var err error
+
+				client, err = getEngineerClient(eksCluster, awsAccountID)
+				framework.ExpectNoError(err)
+			})
+
+			g.It("should allow write access for non-system resources", func() {
+				err := client.RbacV1().ClusterRoles().Delete(context.Background(), nonSystemResource.Name, metav1.DeleteOptions{DryRun: []string{"All"}})
+				framework.ExpectNoError(err, "failed to delete cluster role: %s", nonSystemResource.Name)
+			})
+
+			g.It("should deny write access for system resources", func() {
+				err := client.RbacV1().ClusterRoles().Delete(context.Background(), systemResource.Name, metav1.DeleteOptions{DryRun: []string{"All"}})
+				gomega.Expect(err).To(gomega.MatchError(gomega.ContainSubstring("write operations are forbidden")))
+			})
```

Comment:
Let's add the third test case from `collaborator` here as well to test that it's different, e.g.

```
g.It("should deny write access for collaborator resources", func() {
```

Reply:
Unless that's not what we want. I'm not sure actually.